gatsby-node.js

```diff
@@ -27,6 +27,58 @@ exports.onCreateNode = (gatsbyCreateNodeArgs) => {
   }
 };
 
+/**
+ * We create the preview cards within `onCreateNode`
+ * but we have to tell gatsby about the foreign key relationship
+ * of the linked File.
+ */
+exports.createSchemaCustomization = ({ actions, schema }) => {
+  const { createTypes } = actions;
+
+  const typeDefs = [
+    /**
+     * Variant 1: use `@link` and link the given field
+     * You can't replace fields.socialCard hence the creation of the node in the parent
+     */
+    `type MarkdownRemark implements Node { 
+      socialCard: File @link(from: "fields.socialCard")
+    }`,
+    /**
+     * Variant 2: use `buildObjectType` and resolve the
+     * foreign key relation through a resolver. The `@link` above
+     * is basically syntactic sugar for this. I want to show both variants
+     * here for education purposes as it was a long bumpy road.
+     *
+     * Before ew used the deprecated fielname___NODE way of adding
+     * the FK relationship which fails for incremental builds in gatsby 4
+     *
+     *
+     * Deprecated Note:
+     * https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v3-to-v4/#___node-convention-is-deprecated
+     *
+     * How to "schema additions"
+     * https://www.gatsbyjs.com/docs/how-to/plugins-and-themes/creating-a-source-plugin/#create-foreign-key-relationships-between-data
+     */
+    schema.buildObjectType({
+      name: 'SyPersonioJob',
+      fields: {
+        socialCard: {
+          type: 'File',
+          resolve: (source, args, context, info) => {
+            return context.nodeModel.getNodeById({
+              id: source.fields.socialCard,
+              type: 'File',
+            });
+          },
+        },
+      },
+      interfaces: ['Node'],
+    }),
+  ];
+
+  createTypes(typeDefs);
+};
+
```

Comment:
yeah that solved my hour long trouble getting the file properly connected to the career and blog posts. It always worked on a fresh build but incremental builds failed. Probably because the incremental build hat some issues restoring through the implicit fk `___NODE` assignment we used before. much better because it's explicit now.